posts/2021-08-30-open-liberty-stack-gradle.adoc

```diff
@@ -0,0 +1,96 @@
+
+== OpenShift Do (odo)
+
+link:https://odo.dev[Odo] is a simple CLI tool that you use to create devfile based components that interact directly with your Kubernetes cluster to setup the environment, and build/deploy/access/debug your application. Directives to manage the environment and application are provided by a component's devfile.
+
+== Try it out
+
+To begin, you need three things:
+
+- Kubernetes cluster (i.e. Openshift, Minikube). Be sure to log in. Odo will interact with your kubernetes cluster.
+- Openshift Do. If you have not already installed odo, do so now by following the instructions outlined in the link:https://odo.dev[odo] documentation.
+- An application using Gradle. You can use your own application, or use one of the stack provided applications such as the link:https://github.com/OpenLiberty/application-stack-intro[intro] or link:https://github.com/OpenLiberty/application-stack-starters[starter] application. Instruction on how to use the starter application can be found link:https://github.com/OpenLiberty/application-stack/wiki/Using-the-Default-Starter#getting-started[here]. For this blog, we will use the intro application. This is a basic application that uses JAX-RS and MicroProfile technologies.
+
+Clone the application.
+
+[source,sh]
+----
+$ git clone git@github.com:OpenLiberty/application-stack-intro.git && \
+  cd application-stack-intro
+----
+
+Create an odo component of type java-openliberty-gradle. When the component is created, the Gradle based devfile provided by the Open Liberty devfile stack is downloaded to the root of the application.
+
+[source,sh]
+----
+$ odo create java-openliberty-gradle my-ol-component
+----
+
+Deploy the application on your Kubernetes cluster.
+
+[source,sh]
+----
+$ odo push
+----
+
+That is all. The application is now deployed on your cluster, and a URL has been made available to access the application. 
+
+Notice that you did not have to create any Kubernetes resources to deploy the application. The work to make that happen was done behind the scenes by odo and the stack.
+
+Now, let's see if the application was deployed successfully. For this, we need to get the URL to access the application.
+
+[source,sh]
+----
+$ odo url list
+
+Found the following URLs for component my-ol-component
+NAME     STATE      URL                                                                     PORT     SECURE     KIND
+ep1      Pushed     http://ep1-my-ol-component-my-project.my.kube.cluster.ibm.com           9080     false      route
+----
+
+The default endpoint name assigned by the stack's devfile is ep1.
+
+To access the REST endpoint for the application, append `/api/resource` to the URL listed in the previous step (i.e. `http://ep1-my-ol-component-my-project.my.kube.cluster.ibm.com/api/resource`) and access it through a browser or through the command line.
```

Comment:
suggested edit:

To access the REST endpoint for the application, append `/api/resource` to the URL listed in the previous step (i.e. `http://ep1-my-ol-component-my-project.my.kube.cluster.ibm.com/api/resource`) and access it through a browser or through the command line.

--->

To access the REST endpoint for the application, append `/api/resource` to the URL listed in the previous step, for example, `http://ep1-my-ol-component-my-project.my.kube.cluster.ibm.com/api/resource`. You can access this endpoint through a browser or by running a `curl` command: